Makefile

```diff
@@ -106,6 +106,11 @@ clean:
 	@rm -Rf *.wasm
 	cargo clean
 
+test-pure: FEATURES=mainnet,integration-test,meta-call
+test-pure: Cargo.toml Cargo.lock $(shell find src -name "*.rs") 
+	RUSTFLAGS='-C link-arg=-s' $(CARGO) build --target wasm32-unknown-unknown --release --no-default-features --features=$(FEATURES) -Z avoid-dev-deps
+	$(CARGO) test --features mainnet-test
```

Comment:
It's not run Solidity code.

Reply:
Some rust tests will fail without the solidity artifacts if they do not exist, so this Make target will not work in that case. And in the case the solidity artifacts already exist they are not recompiled, meaning it will behave the same as `test-mainnet` in that case. So I think we should not have this Make target.

Reply:
So let's understand why we need Makefile? For development simplification. For example, to run Node-related code I should run it in a separate environment. But I should run ONLY rust code. So for rust-related source code, I shouldn't use Node dependency. So without that, it's impossible for me to run any tests.

Reply:
We don't have a dedicated command for that in Makefile. So I should have one exactly for cases when I need to test only Rust code.

Reply:
```
$ git status
On branch develop
Your branch is up to date with 'origin/develop'.

$ make test-mainnet
RUSTFLAGS='-C link-arg=-s' cargo build --target wasm32-unknown-unknown --release --no-default-features --features=mainnet,integration-test,meta-call -Z avoid-dev-deps
    Finished release [optimized] target(s) in 0.92s
cp target/wasm32-unknown-unknown/release/aurora_engine.wasm mainnet-release.wasm
cargo test --features mainnet-test
   Compiling aurora-engine v1.0.0 (/datadrive/birchmd/aurora-engine)
    Finished test [optimized] target(s) in 42.02s
     Running unittests (target/debug/deps/aurora_engine-0cb5a3fcb128ded8)

running 88 tests
test benches::benches ... ignored
test json::tests::test_json_type_u128 ... ok
test json::tests::test_json_type_bool ... ok
test json::tests::test_json_type_string ... ok
test json::tests::test_json_type_u64 ... ok
test json::tests::test_json_type_u8 ... ok
test parameters::tests::test_roundtrip_view_call ... ok
test parameters::tests::test_view_call_fail ... ok
test precompiles::blake2::tests::test_blake2f ... ok
test precompiles::bn128::tests::test_alt_bn128_add ... ok
test precompiles::hash::tests::test_ripemd160 ... ok
test precompiles::hash::tests::test_sha256 ... ok
test precompiles::identity::tests::test_identity ... ok
test precompiles::modexp::tests::test_berlin_modexp_big_input ... ok
test precompiles::modexp::tests::test_berlin_modexp_bigger_input ... ok
test precompiles::modexp::tests::test_berlin_modexp_empty_input ... ok
test precompiles::bn128::tests::test_alt_bn128_mul ... ok
test meta_parsing::tests::test_parse_type ... ok
...
```

`test-mainnet` already will only run rust tests (see above), we do not need a new Make target for this use case.

Reply:
You're wrong:
```
$ make test-mainnet 
cd etc/eth-contracts && yarn && yarn build
/bin/sh: 1: yarn: not found
Makefile:36: recipe for target 'etc/eth-contracts/res/EvmErc20.bin' failed
make: *** [etc/eth-contracts/res/EvmErc20.bin] Error 127
```

Reply:
It's impossible to run it without Node.js

Reply:
That should only happen if `EvmErc20.bin` does not exist. And if that is the case then you cannot compile the rust part either because it [imports that file](https://github.com/aurora-is-near/aurora-engine/blob/develop/src/lib.rs#L351).